config/dynamic-config-builder/src/validator_config.rs

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::Error;
+use anyhow::{ensure, Result};
+use libra_config::{
+    config::{ConsensusPeersConfig, NodeConfig, SeedPeersConfig},
+    generator,
+};
+use libra_crypto::{ed25519::Ed25519PrivateKey, PrivateKey, Uniform};
+use libra_types::transaction::Transaction;
+use parity_multiaddr::Multiaddr;
+use rand::{rngs::StdRng, Rng, SeedableRng};
+use std::collections::HashMap;
+use vm_genesis;
+
+const DEFAULT_SEED: [u8; 32] = [13u8; 32];
+const DEFAULT_IPV4_ADVERTISED: &str = "/ip4/127.0.0.1/tcp/6180";
+const DEFAULT_IPV4_LISTEN: &str = "/ip4/0.0.0.0/tcp/6180";
+const DEFAULT_IPV6_ADVERTISED: &str = "/ip6/::1/tcp/6180";
+const DEFAULT_IPV6_LISTEN: &str = "/ip6/::/tcp/6180";
+
+pub struct ValidatorConfig {
+    advertised: Multiaddr,
+    bootstrap: Multiaddr,
+    index: usize,
+    ipv4: bool,
+    listen: Multiaddr,
+    nodes: usize,
+    seed: [u8; 32],
+    template: NodeConfig,
+}
+
+impl Default for ValidatorConfig {
+    fn default() -> Self {
+        Self {
+            advertised: DEFAULT_IPV4_ADVERTISED.parse::<Multiaddr>().unwrap(),
+            bootstrap: DEFAULT_IPV4_ADVERTISED.parse::<Multiaddr>().unwrap(),
+            index: 0,
+            ipv4: true,
```

Comment:
isn't it weird to have ipv4 here since it is already explicit in advertised/bootstrap/listen?

Reply:
that's why I deleted it in the dynamic-config ... there's some weirdness in our code about assumptions that can be cleaned up.